### DISCUSSION TITLE
Implement a feature to import any file extension as an Astro component

### DISCUSSION THREAD

**Discussion by kireerik** [withastro/roadmap#589] [Proposal]
### Body

example:
```Astro
---
import Component from 'name.extension?astro'
---
```

&nbsp;
> The `astro` part could be named `.astro`, `Astro`, `AstroComponent`, `astro-component` and or  `astroComponent`.

&nbsp;
This can be useful for example to inline SVG files.

example:
```Astro
---
import Svg from '.svg?astro'
---

<Svg>
```

**Comment by jasikpark** [comment]
++ to a very scoped version of this -- I love reactsvgr for using svgs as react components, and https://github.com/jasikpark/astro-svg-loader/ was an attempt at replicating that workflow. 

I think a second attempt that would be great is altering the import of an svg file to be similar to an html import - they are both imported as astro fragments

Refs:
- https://github.com/jasikpark/astro-svg-loader/issues/56#issuecomment-1561725569

**Reply by jasikpark** [reply]
I'd be inclined to implement this as a plugin first, rather than go directly through RFC, since this seems like it should be a simple plugin similar to the images addon?

**Reply by kireerik** [reply]
> If we rename an `svg` file's extension from `svg` to `astro` then we can already import it as an Astro component.

So maybe it is simple to implement this feature since there is already a `raw` option for example.

**Comment by kireerik** [comment]
This could be nice, but since the following issue is resolved there is already a solution for this:
- https://github.com/MrFoxPro/astro-svg-components/issues/1#issuecomment-1562177826

Plus of course we can already use SVGs (HTML) as Astro components (by writing the SVGs inside Astro components).
